pkg/broker/broker.go

```diff
@@ -18,45 +18,117 @@ package broker
+	case "mariadb":
+		values = d.Mariadb
+	case "mongodb":
+		values = d.Mongodb
+	case "mysql":
+		values = d.Mysql
+	case "postgresql":
+		values = d.Postgresql
+	case "rabbitmq":
+		values = d.Rabbitmq
+	case "redis":
+		values = d.Redis
+	}
+
+	return values, values != nil
+}
+
+// MinibrokerClient defines the interface of the client the broker operates on
+type MinibrokerClient interface {
+	Init(repoURL string) error
+	ListServices() ([]osb.Service, error)
+	Provision(instanceID, serviceID, planID, namespace string, acceptsIncomplete bool, provisionParams map[string]interface{}) (string, error)
+	Bind(instanceID, serviceID, bindingID string, acceptsIncomplete bool, bindParams map[string]interface{}) (string, error)
+	Unbind(instanceID, bindingID string) error
+	GetBinding(instanceID, bindingID string) (*osb.GetBindingResponse, error)
+	Deprovision(instanceID string, acceptsIncomplete bool) (string, error)
+	LastOperationState(instanceID string, operationKey *osb.OperationKey) (*osb.LastOperationResponse, error)
+	LastBindingOperationState(instanceID, bindingID string) (*osb.LastOperationResponse, error)
+}
+
+// NewBrokerFromOptions is a hook that is called with the Options the program is run
 // with. NewBroker is the place where you will initialize your
 // Broker the parameters passed in.
-func NewBroker(o Options) (*Broker, error) {
+func NewBrokerFromOptions(o Options) (*Broker, error) {
 	klog.V(5).Infof("broker: creating a new broker with options %+v", o)
 	mb := minibroker.NewClient(o.ConfigNamespace, o.ServiceCatalogEnabledOnly)
 	err := mb.Init(o.HelmRepoURL)
 	if err != nil {
 		return nil, err
 	}
 
-	// For example, if your Broker requires a parameter from the command
-	// line, you would unpack it from the Options and set it on the
-	// Broker here.
+	overrideChartParams := &OverrideChartParams{}
+	if len(o.OverrideChartParams) > 0 {
+		data, err := ioutil.ReadFile(o.OverrideChartParams)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to read default chart values file '%q': %w", o.OverrideChartParams, err)
+		}
+		overrideChartParams.LoadYaml(data)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to parse default chart values file '%q': %w", o.OverrideChartParams, err)
+		}
+		klog.V(2).Infof("broker: got default chart values: %#v", overrideChartParams)
+	}
+
+	return NewBroker(mb, o.DefaultNamespace, overrideChartParams), nil
+}
+
+// NewBroker creates a Broker instance with the given dependencies
+func NewBroker(mb MinibrokerClient, defaultNamespace string, overrideChartParams *OverrideChartParams) *Broker {
 	return &Broker{
-		Client:           mb,
-		async:            true,
-		defaultNamespace: o.DefaultNamespace,
-	}, nil
+		Client:              mb,
+		async:               true,
+		defaultNamespace:    defaultNamespace,
+		overrideChartParams: overrideChartParams,
+	}
 }
 
 // Broker provides an implementation of broker.Interface
 type Broker struct {
-	Client *minibroker.Client
+	Client MinibrokerClient
```

Comment:
Since you are already here, `Client` doesn't need to be exported. You will need to fix it up when it's used.

```suggestion
	client MinibrokerClient
```